pysingfel/particlePlacement.py

```diff
@@ -27,38 +32,50 @@ def distribute_particles(particles, beam_focus_radius, jet_radius, sticking=Fals
     N = sum(particles.values()) # total number of particles
     coords = np.zeros((N,3)) # initialize N*3 array
     # generate N*3 random positions
-    if sticking:
-        agg = build_bpca(num_pcles=N, radius=radius_max)
-        reference = np.zeros((1,3))
-        reference[0,0] = beam_focus_radius*np.random.uniform(-1, 1)
-        reference[0,1] = beam_focus_radius*np.random.uniform(-1, 1)
-        reference[0,2] = jet_radius*np.random.uniform(-1, 1)
-        for i in range(1,N):
-            coords[i,0] = agg.pos[i,0]+reference[0,0]
-            coords[i,1] = agg.pos[i,1]+reference[0,1]
-            coords[i,2] = agg.pos[i,2]+reference[0,2]
-        return state, coords 
-    else:
-        for i in range(N):
-            coords[i,0] = beam_focus_radius*np.random.uniform(-1, 1)
-            coords[i,1] = beam_focus_radius*np.random.uniform(-1, 1)
-            coords[i,2] = jet_radius*np.random.uniform(-1, 1)
-        # calculate N*N distance matrix
-        dist_matrix = distance.cdist(coords, coords, 'euclidean')
-        # collision detection check (<2 maxRadius)
-        collision = dist_matrix < 2*radius_max
-        checkList = [collision[i][j] for i in range(N) for j in range(N) if j > i]
-        if any(item == True for item in checkList):
-            distribute_particles(particles, beam_focus_radius, jet_radius)
-        return state, coords
+    for i in range(N):
+        coords[i,0] = beam_focus_radius*np.random.uniform(-1, 1)
+        coords[i,1] = beam_focus_radius*np.random.uniform(-1, 1)
+        coords[i,2] = jet_radius*np.random.uniform(-1, 1)
+    # calculate N*N distance matrix
+    dist_matrix = distance.cdist(coords, coords, 'euclidean')
+    # collision detection check (<2 maxRadius)
+    collision = dist_matrix < 2*radius_max
+    checkList = [collision[i][j] for i in range(N) for j in range(N) if j > i]
+    if any(item == True for item in checkList):
+        distribute_particles(particles, beam_focus_radius, jet_radius)
+    # calculate interaction range
+    R_interaction = np.sqrt(beam_focus_radius**2+beam_focus_radius**2+jet_radius**2)*gamma
```

Comment:
should check whether jet_radius is bigger than beam_focus

Reply:
beam_focus_radius = 10e-6, jet_radius = 1e-4 (default)

Reply:
x coordinate = beam_focus_radius*np.random.uniform(-1, 1)
y coordinate = beam_focus_radius*np.random.uniform(-1, 1)
z coordinate = jet_radius*np.random.uniform(-1, 1)
Let R = np.sqrt(beam_focus_radius**2+beam_focus_radius**2+jet_radius**2) is to assume having a spherical force field.

Reply:
1) you shouldn't use default values. It's different for every experiment.
2) you have to check jet_radius is bigger than beam_focus.